### PR TITLE
node/p2p: Deduplicate PubSub msgs

### LINKED ISSUE #236: service/header: deduplicate Header msgs in pubsub

### PR DESCRIPTION
**Pull request by Wondertan**
Closes #236

### CODE REVIEW

**Review by renaynay — APPROVED** [review]

**Review by renaynay — CHANGES_REQUESTED** [review]
lint

**Review by renaynay — APPROVED** [review]

**Review by adlerjohn — APPROVED** [review]
ahh I'm approoooving

File: node/p2p/pubsub.go [hunk 1]
```diff
@@ -29,6 +31,7 @@ func PubSub(cfg Config) func(pubSubParams) (*pubsub.PubSub, error) {
 		opts := []pubsub.Option{
 			pubsub.WithPeerExchange(cfg.PeerExchange || cfg.Bootstrapper),
 			pubsub.WithDirectPeers(fpeers),
+			pubsub.WithMessageIdFn(hashMsgID),
```
Comment (liamsi) [inline node/p2p/pubsub.go:34]:
```suggestion
			pubsub.WithMessageIDFn(hashMsgID),
```

Comment (Wondertan) [reply node/p2p/pubsub.go:34]: This should be transferred to pubsub. The func is not defined by us

**Pull request merged**